src/app/applications/time_lock/page.mdx

```diff
@@ -0,0 +1,372 @@
+export const metadata = {
+  title: "Time Lock • Stylus by Example",
+  description:
+    "An example implementation of the Timelock wallet in Rust using Arbitrum Stylus.",
+};
+
+{/* Begin Content */}
+
+# Timelock Wallet
+
+An Arbitrum Stylus version implementation of [Solidity TimeLock](https://solidity-by-example.org/app/time-lock/).
+
+- TimeLock is a contract that publishes a transaction to be executed in the future. After a minimum waiting period, the transaction can be executed.
+- TimeLocks are commonly used in DAOs.
+
+Here is the interface for TimeLock.
+
+```solidity
+interface ITimeLock {
+    function initialize() external;
+
+    function owner() external view returns (address);
+
+    function getTxId(address target, uint256 value, string calldata func, bytes calldata data, uint256 timestamp) external view returns (bytes32);
+
+    function deposit() external payable;
+
+    function queue(address target, uint256 value, string calldata func, bytes calldata data, uint256 timestamp) external;
+
+    function execute(address target, uint256 value, string calldata func, bytes calldata data, uint256 timestamp) external;
+
+    function cancel(address target, uint256 value, string calldata func, bytes calldata data, uint256 timestamp) external;
+
+    error AlreadyInitialized();
+
+    error NotOwnerError();
+
+    error AlreadyQueuedError(bytes32);
+
+    error TimestampNotInRangeError(uint256, uint256);
+
+    error NotQueuedError(bytes32);
+
+    error TimestampNotPassedError(uint256, uint256);
+
+    error TimestampExpiredError(uint256, uint256);
+
+    error TxFailedError();
+}
+```
+
+Example implementation of a Timelock Wallet contract written in Rust.
```

Comment:
Not sure 100%. Is it called a "Timelock Wallet contract" or just a "Timelock contract"?

Reply:
Sure, we should call it "Timelock contract" directly